common/conf_test.cpp

```diff
@@ -35,3 +36,221 @@ TEST(ConfTests, GetEnvTest) {
+		"arg1",
+		"--opt4=val4",
+	};
+
+	conf::CmdlineOptionsIterator opts_iter(
+		args, {"--opt1", "-o2", "--opt4", "-o6"}, {"--opt3", "--opt5", "-o7"});
+	auto ex_opt_val = opts_iter.Next();
+	ASSERT_TRUE(ex_opt_val);
+	auto opt_val = ex_opt_val.value();
+	EXPECT_EQ(opt_val.option, "--opt1");
+	EXPECT_EQ(opt_val.value, "val1");
+
+	ex_opt_val = opts_iter.Next();
+	ASSERT_TRUE(ex_opt_val);
+	opt_val = ex_opt_val.value();
+	EXPECT_EQ(opt_val.option, "-o2");
+	EXPECT_EQ(opt_val.value, "val2");
+
+	ex_opt_val = opts_iter.Next();
+	ASSERT_TRUE(ex_opt_val);
+	opt_val = ex_opt_val.value();
+	EXPECT_EQ(opt_val.option, "--");
+	EXPECT_EQ(opt_val.value, "");
+
+	ex_opt_val = opts_iter.Next();
+	ASSERT_TRUE(ex_opt_val);
+	opt_val = ex_opt_val.value();
+	EXPECT_EQ(opt_val.option, "");
+	EXPECT_EQ(opt_val.value, "--opt3");
+
+	ex_opt_val = opts_iter.Next();
+	ASSERT_TRUE(ex_opt_val);
+	opt_val = ex_opt_val.value();
+	EXPECT_EQ(opt_val.option, "");
+	EXPECT_EQ(opt_val.value, "arg1");
+
+	ex_opt_val = opts_iter.Next();
+	ASSERT_TRUE(ex_opt_val);
+	opt_val = ex_opt_val.value();
+	EXPECT_EQ(opt_val.option, "");
+	EXPECT_EQ(opt_val.value, "--opt4=val4");
+
+	// termination object
+	ex_opt_val = opts_iter.Next();
+	ASSERT_TRUE(ex_opt_val);
+	opt_val = ex_opt_val.value();
+	EXPECT_EQ(opt_val.option, "");
+	EXPECT_EQ(opt_val.value, "");
+
+	// should stay at the termination object and not fail
+	ex_opt_val = opts_iter.Next();
+	ASSERT_TRUE(ex_opt_val);
+	opt_val = ex_opt_val.value();
+	EXPECT_EQ(opt_val.option, "");
+	EXPECT_EQ(opt_val.value, "");
+}
+
+TEST(ConfTests, CmdlineOptionsIteratorBadOptionTest) {
+	vector<string> args = {
+		"--opt1",
+		"val1",
+		"-o2",
+	};
+
+	conf::CmdlineOptionsIterator opts_iter(
+		args, {"--opt1", "--opt4", "-o6"}, {"--opt3", "--opt5", "-o7"});
+	auto ex_opt_val = opts_iter.Next();
+	ASSERT_TRUE(ex_opt_val);
+	auto opt_val = ex_opt_val.value();
+	EXPECT_EQ(opt_val.option, "--opt1");
+	EXPECT_EQ(opt_val.value, "val1");
+
+	ex_opt_val = opts_iter.Next();
+	ASSERT_FALSE(ex_opt_val);
+	EXPECT_EQ(ex_opt_val.error().message, "Unrecognized option '-o2'");
+}
+
+TEST(ConfTests, CmdlineOptionsIteratorOptionMissingValueTest) {
+	vector<string> args = {
+		"--opt1",
+		"-o2",
+		"val2",
+	};
+
+	conf::CmdlineOptionsIterator opts_iter(
+		args, {"--opt1", "-o2", "--opt4", "-o6"}, {"--opt3", "--opt5", "-o7"});
+	auto ex_opt_val = opts_iter.Next();
+	ASSERT_FALSE(ex_opt_val);
+	EXPECT_EQ(ex_opt_val.error().message, "Option --opt1 missing value");
+}
+
+TEST(ConfTests, CmdlineOptionsIteratorOptionMissingValueTrailingTest) {
+	vector<string> args = {
+		"--opt1",
+	};
+
+	conf::CmdlineOptionsIterator opts_iter(
+		args, {"--opt1", "-o2", "--opt4", "-o6"}, {"--opt3", "--opt5", "-o7"});
+	auto ex_opt_val = opts_iter.Next();
+	ASSERT_FALSE(ex_opt_val);
+	EXPECT_EQ(ex_opt_val.error().message, "Option --opt1 missing value");
+}
+
+TEST(ConfTests, CmdlineOptionsIteratorOptionExtraValueTest) {
+	vector<string> args = {
+		"--opt3=val3",
+		"-o2",
```

Comment:
I don't remember what the conclusion was, but I thought we had to support `-o 2` also?

Reply:
I think it does, it actually is doing `-o2 val1`. Using single-dash multi-letter options is not something are going to use, but perfectly fine that it supports it.